python/runtime/utils/PyRestRemoteClient.cpp

```diff
@@ -79,29 +79,49 @@ class PyNvcfRuntimeClient : public PyRestRemoteClient {
     return versions;
   }
   DeploymentInfo getAllAvailableDeployments() {
-    DeploymentInfo info;
     auto headers = getHeaders();
     auto allVisibleFunctions =
         m_restClient.get(fmt::format("https://{}/nvcf/functions", m_baseUrl),
                          "", headers, /*enableSsl=*/true);
-    for (auto funcInfo : allVisibleFunctions["functions"]) {
-      if (funcInfo["ncaId"].get<std::string>() == CUDAQ_NCA_ID &&
-          funcInfo["status"].get<std::string>() == "ACTIVE" &&
-          funcInfo["name"].get<std::string>().starts_with("cuda_quantum")) {
-        const std::size_t numGpus = [&]() {
-          if (funcInfo.contains("containerEnvironment")) {
-            for (auto it : funcInfo["containerEnvironment"]) {
-              if (it["key"].get<std::string>() == "NUM_GPUS")
-                return std::stoi(it["value"].get<std::string>());
+    const auto queryDeploymentInfo =
+        [&](const std::string &functionNamePrefix) -> DeploymentInfo {
+      DeploymentInfo info;
+      for (auto funcInfo : allVisibleFunctions["functions"]) {
+        if (funcInfo["ncaId"].get<std::string>() == CUDAQ_NCA_ID &&
+            funcInfo["status"].get<std::string>() == "ACTIVE" &&
+            funcInfo["name"].get<std::string>().starts_with(
+                functionNamePrefix)) {
+          const std::size_t numGpus = [&]() {
+            if (funcInfo.contains("containerEnvironment")) {
+              for (auto it : funcInfo["containerEnvironment"]) {
+                if (it["key"].get<std::string>() == "NUM_GPUS")
+                  return std::stoi(it["value"].get<std::string>());
+              }
             }
-          }
-          return 1;
-        }();
+            return 1;
+          }();
 
-        info[funcInfo["id"].get<std::string>()] = numGpus;
+          info[funcInfo["id"].get<std::string>()] = numGpus;
+        }
       }
-    }
-    return info;
+      return info;
+    };
+
+    const std::string baseCudaqNvcfFuncNamePrefix = "cuda_quantum";
+    const std::string versionedCudaqNvcfFuncNamePrefix =
+        baseCudaqNvcfFuncNamePrefix + "_" +
+        std::to_string(cudaq::RestRequest::REST_PAYLOAD_VERSION);
```

Comment:
It would be best to add an underscore at the end so that when `REST_PAYLOAD_VERSION` = 1 that it does not match `cuda_quantum_11`.

```suggestion
    const std::string versionedCudaqNvcfFuncNamePrefix =
        baseCudaqNvcfFuncNamePrefix + "_" +
        std::to_string(cudaq::RestRequest::REST_PAYLOAD_VERSION) + "_";
```

Reply:
Good catch. Thanks!